File: backend/src/main/java/com/novelwriting/entity/Outline.java
package com.novelwriting.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Data
@Entity
@Table(name = "outlines")
public class Outline {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "novel_id")
    private Long novelId;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(name = "chapter_number")
    private Integer chapterNumber;

    @Column(name = "plot_order")
    private Integer plotOrder;

    @Column(length = 20)
    private String status = "planning";

    @ManyToMany
    @JoinTable(
        name = "outline_tags",
        joinColumns = @JoinColumn(name = "outline_id"),
        inverseJoinColumns = @JoinColumn(name = "tag_id")
    )
    private Set<Tag> tags = new HashSet<>();

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend-vue/src/types/outline.ts
import type { ID, Timestamp } from './index'
import type { Tag } from './tag'

export type OutlineStatus = 'planning' | 'writing' | 'completed'

export interface Outline {
  id: ID
  title: string
  content?: string
  status: OutlineStatus
  chapterNumber?: number
  plotOrder?: number
  tags?: Tag[]
  createdAt?: Timestamp
  updatedAt?: Timestamp
}

export interface OutlineCreateDTO {
  title: string
  content?: string
  status?: OutlineStatus
  chapterNumber?: number
  plotOrder?: number
}

export interface OutlineUpdateDTO extends Partial<OutlineCreateDTO> {}
